daemons/fenced/fenced_commands.c

```diff
@@ -900,11 +900,17 @@ xml2device_params(const char *name, xmlNode *dev)
 static stonith_device_t *
 build_device_from_xml(xmlNode * msg)
 {
-    const char *value = NULL;
+    const char *value;
     xmlNode *dev = get_xpath_object("//" F_STONITH_DEVICE, msg, LOG_ERR);
     stonith_device_t *device = NULL;
+    char *agent = crm_element_value_copy(dev, "agent");
+
+    CRM_CHECK(agent != NULL, return device);
 
     device = calloc(1, sizeof(stonith_device_t));
+
+    CRM_CHECK(device != NULL, {free(agent); return device;});
```

Comment:
These brackets are unnecessary (brackets are part of the macro). It's fine to leave them for readability if that's your goal here, though that isn't done elsewhere.

Reply:
Yes, feels more clear this way, separators are easy to get wrong.